tequilapi/endpoints/service.go

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package endpoints
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/mysteriumnetwork/node/core/service"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/tequilapi/utils"
+	"github.com/mysteriumnetwork/node/tequilapi/validation"
+)
+
+// swagger:model ServiceRequestDTO
+type serviceRequest struct {
+	// provider identity
+	// required: true
+	// example: 0x0000000000000000000000000000000000000002
+	ProviderID string `json:"providerId"`
+
+	// service type. Possible values are "openvpn", "wireguard" and "noop"
+	// required: false
+	// default: openvpn
+	// example: openvpn
+	ServiceType string `json:"serviceType"`
+}
+
+// swagger:model ServiceListDTO
+type serviceList []serviceInfo
+
+// swagger:model ServiceInfoDTO
+type serviceInfo struct {
+	Proposal proposalRes `json:"proposal"`
+	// example: Running
+	Status  string         `json:"status"`
+	Options serviceOptions `json:"options"`
+}
+
+type serviceOptions struct {
+	// example: UDP
+	Protocol string `json:"protocol"`
+	// example: 1190
+	Port int `json:"port"`
+}
+
+// ServiceEndpoint struct represents /service resource and it's sub-resources
+type ServiceEndpoint struct {
+	serviceManager ServiceManager
+	serviceState   service.State
+}
+
+// NewServiceEndpoint creates and returns service endpoint
+func NewServiceEndpoint(serviceManager ServiceManager) *ServiceEndpoint {
+	return &ServiceEndpoint{
+		serviceManager: serviceManager,
+		serviceState:   service.NotRunning,
+	}
+}
+
+// ServiceList provides a list of running services on the node.
+// swagger:operation GET /services Service serviceList
+// ---
+// summary: List of services
+// description: ServiceList provides a list of running services on the node.
+// responses:
+//   200:
+//     description: List of running services
+//     schema:
+//       "$ref": "#/definitions/ServiceListDTO"
+func (se *ServiceEndpoint) ServiceList(resp http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+	statusResponse := toServiceListResponse(se.serviceState)
+	utils.WriteAsJSON(statusResponse, resp)
+}
+
+// ServiceGet provides info for requested service on the node.
+// swagger:operation GET /services/:id Service serviceGet
```

Comment:
It's a bit unclear what is service id - how to get it?

Reply:
ID is a unique identifier for the service. It can be get from the list of services `GET /services`

There are two possible options that I see now: 

1. Something like UUID just to have a unique identifier;
1. Service type like `openvpn`, `wireguard`, `noop`. Since there is no point to provide more than one services of the particular type.

@mysteriumnetwork/core what do you think about this ID, what should we use? 

I personally like the second option more if we are really not planning to provide several services of the single type from one node.

Reply:
> ID is a unique identifier for the service. It can be get from the list of services `GET /services`

But I don't see any id in `serviceInfo` struct, which is returned from `GET /services`:
```go
	type serviceInfo struct {
	Proposal proposalRes `json:"proposal"`
	// example: Running
	Status  string         `json:"status"`
	Options serviceOptions `json:"options"`
}
```

> Service type like `openvpn`, `wireguard`, `noop`
This makes sense, but in that case `ServiceRequestDTO` should not require `providerID` as well - ID should be consistent across different endpoints.

Reply:
`serviceInfo` includes a proposal, which has both `providerID` and `ServiceType`. 

For the starting service, `ServiceRequestDTO` requires `providerID` to define under which identity service should be started. We cannot skip it.

Reply:
I dont think we should mix `proposalId` with `serviceInstanceId`

Reply:
The question is do we need serviceInstanceID at all, or it’s enough to have only serviceType as identifier?

Reply:
We talked about that  with @zolia. No point to add any limitations from technical side:
- if several Openvpn services can be started, why do deny it
- if we decide to limit that, we should do it in `serviceCreate` validators

Reply:
Added mocked ID for services